File: Project2/Heracleum.cpp
#include "Heracleum.h"
#include "World.h"
#include <stdlib.h>

Heracleum::Heracleum(const Vector2& pos) : Plant(pos, 10, 10, '%')
{
}

Heracleum::~Heracleum()
{
}

void Heracleum::Action()
{
	Plant::Action();

	for (int i = 0; i < 8; i++)
	{
		Vector2 target = pos.GetNeighbor8Way(i);
		Animal* animal = dynamic_cast<Animal*>(world->GetOrganism(target));
		if (animal != nullptr)
		{
			animal->Die(this);
		}
	}
}

Heracleum* Heracleum::Clone(const Vector2& pos) const
{
	return new Heracleum(pos);
}

std::string Heracleum::GetName() const
{
	return "Heracleum";
}

File: Project2/WorldSimulator/Plants/Heracleum.java
package WorldSimulator.Plants;

import WorldSimulator.Animal;
import WorldSimulator.Organism;
import WorldSimulator.Plant;
import WorldSimulator.Vector2;

import java.awt.*;
import java.util.ArrayList;

public class Heracleum extends Plant
{
    public Heracleum(Vector2 pos)
    {
        super(pos, 10, 10, '%', new Color(181, 213, 41));
    }

    @Override
    public void Action()
    {
        super.Action();

        ArrayList<Vector2> targets = world.GetNeighborsWithDiagonals(pos);
        for (Vector2 target : targets)
        {
            Organism org = world.GetOrganism(target);
            if (org instanceof Animal)
            {
                org.Die(this);
            }
        }
    }

    @Override
    public String toString()
    {
        return "Heracleum";
    }
}
